app/controllers/verify/finance_controller.rb

```diff
@@ -17,20 +19,39 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_finance_form: idv_finance_form,
+        idv_session: idv_session
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
+
     def render_form
-      if finance_params[:finance_type] == 'ccn'
+      if step_params[:finance_type] == 'ccn'
         render :new
       else
         render 'verify/finance_other/new'
       end
     end
 
-    def idv_finance_form
-      @_idv_finance_form ||= Idv::FinanceForm.new(idv_session.params)
+    def track_idv_event
+      result = {
+        success: idv_session.financials_confirmation.success?
+      }
+
+      analytics.track_event(Analytics::IDV_FINANCE_CONFIRMATION, result)
```

Comment:
Thanks for refactoring. What I had in mind was writing the code in such a way to make it easy to capture more details about the form submission, such as any field validation errors. With the code here, the `track_idv_event` method will only be called if the form validations passed, so it won't be possible to capture form validation errors, unless a separate analytics event is created, which we want to avoid since each form submission is one event.

What I was envisioning was a change similar to #931, where the `Idv::FinanceForm` would be updated to return a result hash that included whether or not the form was submitted successfully, and any form errors. To capture the vendor validation result, it would be added as a new validation to the Form. The controller would only make one call to the Form, such as:
```ruby
result = Idv::FinanceForm.new(idv_session.params).submit(step_params)
analytics.track_event(Analytics::IDV_FINANCE_CONFIRMATION, result)
if result[:success]
  handle_valid_form_submission
else
  handle_failed_form_submission
end
```
This would allow us to capture both form and vendor validation errors in one place.

The vendor validation itself could/should be defined in a separate class, and called by the form. The Form would need accept the `idv_session` as an argument since the vendor validation needs that. 

For example: https://github.com/18F/identity-idp/commit/a0f7621c4fd0da1ec761baf6efe7a6454e41f654

I'd be happy to pair on this tomorrow if you want.